crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -290,6 +298,36 @@ def ha_config_check
             end
           end
         end
+
+        deployment_supported =
+          case clusters_roles.length
+          when 0
+            # no clusters, no point complaining as this will be detected by other prechecks
+            true
+
+          when 1
+            # everything on one cluster = no problem
+            true
+
+          when 2
+            # neutron-network in separate cluster
+            true if clusters_roles[roles_clusters["neutron-network"]].length == 1 ||
+                # neutron-network + neutron-server in separate cluster
+                (clusters_roles[roles_clusters["neutron-network"]].length == 2 &&
+                roles_clusters["neutron-network"] == roles_clusters["neutron-server"]) ||
+                # database-server + rabbitmq-server in separate cluster
+                (clusters_roles[roles_clusters["database-server"]].length == 2 &&
+                roles_clusters["database-server"] == roles_clusters["rabbitmq-server"])
+
+          when 3
+            # neutron-network and database-server + rabbitmq-server in separate clusters
+            # rest of *-server roles is implicitly on the third cluster
+            true if clusters_roles[roles_clusters["neutron-network"]].length == 1 &&
+                clusters_roles[roles_clusters["database-server"]].length == 2 &&
+                roles_clusters["database-server"] == roles_clusters["rabbitmq-server"]
+          end
+        ret[:unsupported_cluster_setup] = true unless deployment_supported
```

Comment:
As the `case` is a whitelist I don't see how I can deduce any error information from this.